### PR TITLE
TST: Improved test (getting rid of image comparison test for test_titletwiny)

### PR DESCRIPTION
**Pull request by pharshalp**
## PR Summary
attempting to address the second bullet in #15054  🤞
(getting rid of image comparison test for test_titletwiny)

will create separate PRs for the other two tests.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -6014,18 +6014,20 @@ def test_axisbelow():
...
     ax2 = ax.twiny()
-    ax.set_xlabel('Xlabel')
-    ax2.set_xlabel('Xlabel2')
-    ax.set_title('Title')
+    xlabel2 = ax2.set_xlabel('Xlabel2')
+    title = ax.set_title('Title')
+    fig.canvas.draw()
+    renderer = fig.canvas.get_renderer()
+    # ------- Test that title is put above Xlabel2 (Xlabel2 at top) ----------
+    Bbox_y0_title = title.get_window_extent(renderer).y0  # bottom of title
```
Comment (anntzer) [inline lib/matplotlib/tests/test_axes.py]: capitalization makes github highlight this weirdly

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -6014,18 +6014,20 @@ def test_axisbelow():
...
+    Bbox_y1_xlabel2 = xlabel2.get_window_extent(renderer).y1  # top of xlabel2
+    ydiff_px = Bbox_y0_title-Bbox_y1_xlab_top  # difference in pixel
```
Comment (anntzer) [inline lib/matplotlib/tests/test_axes.py]: normal style would have spaces around operators (below too)

**Comment by anntzer** [comment]
Just a few nits, but otherwise lgtm, thanks for tacking this :)

**Comment by pharshalp** [comment]
interestingly the test passes when I run it on its own... need to take a closer look at my approach. may be I am missing something very fundamental...

File: lib/matplotlib/tests/test_axes.py [hunk 3]
```diff
@@ -6014,18 +6014,19 @@ def test_axisbelow():
...
-    ax.set_title('Title')
+    xlabel2 = ax2.set_xlabel('Xlabel2')
+    title = ax.set_title('Title')
+    fig.canvas.draw()
+    renderer = fig.canvas.get_renderer()
+    # ------- Test that title is put above Xlabel2 (Xlabel2 at top) ----------
+    bbox_y0_title = title.get_window_extent(renderer).y0  # bottom of title
+    bbox_y1_xlabel2 = xlabel2.get_window_extent(renderer).y1  # top of xlabel2
+    y_diff = bbox_y0_title - bbox_y1_xlabel2
+    assert round(y_diff, 6) == 3.0
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_axes.py]: Should this be tested via [pytest.approx](https://docs.pytest.org/en/latest/reference.html#pytest-approx)?

Comment (pharshalp) [reply lib/matplotlib/tests/test_axes.py]: done.

File: lib/matplotlib/tests/test_axes.py [hunk 4]
```diff
@@ -6014,18 +6014,19 @@ def test_axisbelow():
...
-    ax.set_title('Title')
+    xlabel2 = ax2.set_xlabel('Xlabel2')
+    title = ax.set_title('Title')
+    fig.canvas.draw()
+    renderer = fig.canvas.get_renderer()
+    # ------- Test that title is put above Xlabel2 (Xlabel2 at top) ----------
+    bbox_y0_title = title.get_window_extent(renderer).y0  # bottom of title
+    bbox_y1_xlabel2 = xlabel2.get_window_extent(renderer).y1  # top of xlabel2
+    y_diff = bbox_y0_title - bbox_y1_xlabel2
+    assert y_diff == pytest.approx(3)
```
Comment (jklymak) [inline lib/matplotlib/tests/test_axes.py]: Sorry to be a PITA, but I think its useful to be consistent for folks.  Don't we use `np.testing.allclose` a lot more than `purest.approx`?

Comment (pharshalp) [reply lib/matplotlib/tests/test_axes.py]: yes, across the test suite it looks like `pytest.approx` has only been used thrice (excluding this PR).
Will change it to `assert np.isclose(y_diff, 3)`.

Comment (timhoffm) [reply lib/matplotlib/tests/test_axes.py]: 👍 sorry for the noise.

**Pull request merged**